modules/tensor_mechanics/src/materials/ADHillElastoPlasticityStressUpdate.C

```diff
@@ -201,16 +201,40 @@ ADHillElastoPlasticityStressUpdate::computeElasticityTensorEigenDecomposition()
     for (unsigned int index_j = 0; index_j < dimension; index_j++)
       B_eigen(index_i, index_j) = MetaPhysicL::raw_value(b_matrix(index_i, index_j));
 
-  Eigen::SelfAdjointEigenSolver<AnisotropyMatrixReal> es_b(B_eigen);
-
-  auto lambda_b = es_b.eigenvalues();
-  auto v_b = es_b.eigenvectors();
-  for (unsigned int index_i = 0; index_i < dimension; index_i++)
-    _b_eigenvalues[_qp](index_i) = lambda_b(index_i);
-
-  for (unsigned int index_i = 0; index_i < dimension; index_i++)
-    for (unsigned int index_j = 0; index_j < dimension; index_j++)
-      _b_eigenvectors[_qp](index_i, index_j) = v_b(index_i, index_j);
+  Eigen::SelfAdjointEigenSolver<AnisotropyMatrixRealBlock> es_b_left(B_eigen.block<3, 3>(0, 0));
+  Eigen::SelfAdjointEigenSolver<AnisotropyMatrixRealBlock> es_b_right(B_eigen.block<3, 3>(3, 3));
```

Comment:
isn't that block diagonal anyways?

Reply:
Not in this case. Here, `B_eigen` is Hill tesnor multiplied with the eigen vectors of the elasticity tensor. For the test cases, the top left part was rather a diagonal matrix. But that may not be always true. So, I did the decomposition for both diagonal blocks.

Reply:
But `B_eigen.block<3, 3>(3, 0)` and `B_eigen.block<3, 3>(0, 3)` are still 0?

Reply:
Yes, they are. `B_eigen` still is block diagonal.

Reply:
The test cases for this class `ADHillElastoPlasticityStressUpdate` are not exhaustive and may be overlooking matrix structures that can happen in practice. I'd suggest checking that the assumption is right (hopefully not too compute intensive). If we don't do that, this simplification may lead to wrong results which may be hard to catch in the future.

Reply:
Done!